### PR TITLE
[PLAT-73030] Verify wildcard filter won't match metric without the label

### PR DESCRIPTION
**Pull request by jnyi**
## How did we test
Ran the added unit test and verified it won't match metrics without the label for wildcard

```
~/repos/m3 (filter-tests) $ go test -run '.*TestNonexistTagsFilterMatches.*' github.com/m3db/m3/src/metrics/filters
ok  	github.com/m3db/m3/src/metrics/filters	1.080s
```

**What this PR does / why we need it**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing and/or backwards incompatible change?**:

```release-note

```

**Does this PR require updating code package or user-facing documentation?**:

```documentation-note

```

### CODE REVIEW

File: src/metrics/filters/tags_filter_test.go [hunk 1]
```diff
@@ -88,6 +88,26 @@ func TestEmptyTagsFilterMatches(t *testing.T) {
 	require.True(t, matches)
 }
 
+func TestNonexistTagsFilterMatches(t *testing.T) {
+	filters := map[string]FilterValue{
+		"tagName1": FilterValue{Pattern: "tagValue1"},
```
Comment (hczhu) [inline src/metrics/filters/tags_filter_test.go:93]: Can we also add another test case for duplicate tag keys and values, since we are going to append tags to the original filter string?
E.g. 
```
"filter": "__name__:logdaemon_aggregatedRate project:* proxy:kinesis2prom project:*"

"filter": "__name__:logdaemon_aggregatedRate project:log-daemon proxy:kinesis2prom project:*"
```

**Review by hczhu — COMMENTED** [review]
image

**Review by hczhu — APPROVED** [review]

File: src/metrics/filters/tags_filter_test.go (hunk 1 above)
Comment (jnyi) [reply src/metrics/filters/tags_filter_test.go:93]: No, i won't append if `project` already exists
